lib/pages/chat_list/receive_sharing_intent_mixin.dart

```diff
@@ -35,9 +36,32 @@ mixin ReceiveSharingIntentMixin<T extends StatefulWidget> on State<T> {
           },
         )
         .toList();
-    TwakeApp.router.go('/share');
+    openSharePage();
   }
 
+  void openSharePage() {
+    if (isCurrentPageIsNotRooms()) {
+      return;
+    }
+    if (isCurrentPageIsInRooms()) {
+      TwakeApp.router.go('/rooms');
+    }
+
+    Navigator.of(TwakeApp.routerKey.currentContext!).push(
+      MaterialPageRoute(
+        builder: (context) => const Share(),
+      ),
+    );
+  }
+
+  bool isCurrentPageIsInRooms() =>
+      TwakeApp.router.routeInformationProvider.value.uri.path
+          .startsWith('/rooms/');
+
+  bool isCurrentPageIsNotRooms() =>
+      !TwakeApp.router.routeInformationProvider.value.uri.path
+          .startsWith('/rooms');
```

Comment:
```
  bool isCurrentPageIsNotRooms() =>
      !TwakeApp.router.routeInformationProvider.value.uri.path
          .startsWith('/rooms/');
```

What is the different for this?

Reply:
All pages that are not start with `/rooms`, for example: `/rooms`, and `/twakeWelcome` will have isCurrentPageIsNotRooms is true